com.unity.render-pipelines.universal/Runtime/LightCookieManager.cs

```diff
@@ -70,10 +73,121 @@ public static Settings GetDefault()
             }
         }
 
+        private struct Sorting
+        {
+            public static void QuickSort<T>(T[] data, Func<T, T, int> compare)
+            {
+                QuickSort<T>(data, 0, data.Length - 1, compare);
+            }
+
+            // A non-allocating predicated sub-array quick sort.
+            // NOTE: Similar to UnityEngine.Rendering.CoreUnsafeUtils.QuickSort in CoreUnsafeUtils.cs,
+            // se should see if these could be merged in the future.
+            // For example: Sorting.QuickSort(test, 0, test.Length - 1, (int a, int b) => a - b);
+            public static void QuickSort<T>(T[] data, int start, int end, Func<T, T, int> compare)
+            {
+                int diff = end - start;
+                if (diff < 1)
+                    return;
+                if (diff < 8)
+                {
+                    InsertionSort(data, start, end, compare);
+                    return;
+                }
+
+                Assertions.Assert.IsTrue((uint)start < data.Length);
+                Assertions.Assert.IsTrue((uint)end   < data.Length); // end == inclusive
```

Comment:
Insertion sort can be used separately, that's why it's like that. 
It's fully standalone sort and I've got plans to move it to core if we merge these, since I've seen couple one-off implementations in the code base.

Reply:
To me it still feels odd to have generic sorting algorithms living in SRP Core. Are we sure there are none in the engine itself? We should avoid having different implementations of the same thing in multiple places. Also if this is general purpose use we should really have extensive test coverage.